docs/explanation/e-hardening.md

```diff
@@ -0,0 +1,119 @@
+
+### Juju 
+
+Juju is the component responsible for orchestrating the entire lifecycle, from deployment to Day 2 operations, of 
+all applications. Therefore, it is imperative that it is set up securely. Please refer to the Juju documentation for more information on:
+
+* [Juju security](https://discourse.charmhub.io/t/juju-security/15684)
+* [How to harder your deployment](https://juju.is/docs/juju/harden-your-deployment)
+
+#### Cloud credentials
+
+When configuring the cloud credentials to be used with Juju, ensure that the users have correct permissions to operate at the required level. 
+Juju superusers responsible for bootstrapping and managing controllers require elevated permissions to manage several kind of resources, such as
+virtual machines, networks, storages, etc. Please refer to the references below for more information on the policies required to be used depending on the cloud. 
+
+| Cloud     | Cloud user policies                                                                                                                                                                                                                            |
+|-----------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| OpenStack |                                                                                                                                                                                                                                                |
+| AWS       | [Juju AWS Permission](https://discourse.charmhub.io/t/juju-aws-permissions/5307), [AWS Instance Profiles](https://discourse.charmhub.io/t/using-aws-instance-profiles-with-juju-2-9/5185), [Juju on AWS](https://juju.is/docs/juju/amazon-ec2) | 
+| Azure     | [Juju Azure Permission](https://juju.is/docs/juju/microsoft-azure), [How to use Juju with Microsoft Azure](https://discourse.charmhub.io/t/how-to-use-juju-with-microsoft-azure/15219)                                                                                                  |
+
+#### Juju users
+
+It is very important that the different juju users are set up with minimal permission depending on the scope for their operations. 
+Please refer to the [User access levels](https://juju.is/docs/juju/user-permissions) documentation for more information on the access level and corresponding abilities 
+that the different users can be granted. 
+
+Juju user credentials must be stored securely and rotated regularly to prevent leakage and unauthorized access. 
+
+## Applications
+
+In the following we provide guidance on how to harden your deployment using:
+
+1. Operating System
```

Comment:
I'm not sure what you mean with the host machine. People could be running everything on the cloud, and not having a host. I believe it is sensible to have the OS in the applications, since these are connected to the Kafka and ZooKeeper deployments, and for instance could be even different between different charms. 

We could have a section on "Machine", but these are tightly connected and spawn by juju, not the environment that needs to be set.

Reply:
Yes, this topic can be confusing, but it is really important to set the borders of our product here. It's usually a very fine line, as it's about layers of software and responsibility. I assumed our product is the charmed operator/bundle and everything else, including snaps, images, Juju, etc. is the environment. I understand that it is different in Juju, where each application can be deployed on its own OS image. Do we consider the OS image to be a part of our distribution/bundle or merely a dependency?
I recommend discussing this particular issue with our Security team later. For now, I'm happy to proceed without changes.